### ISSUE TITLE
On First boot no myMPDos apks are installed

### ISSUE THREAD

**Issue by keithmillar** [jcorporation/myMPDos#44]
The Boot log shows some failures one being unable to select packages. I wood like to get this working as it is a fantastic program.

myMPDos 1.6.0 bootstrap script
```
 *   WARNING: you are stopping a sysinit service
 *   Executing: /usr/libexec/rc/sh/openrc-run.sh /usr/libexec/rc/sh/openrc-run.sh /etc/init.d/modloop stop
 *   Unmounting /.modloop ... [ ok ]
Creating data partition on sd card

Command (m for help): The number of cylinders for this disk is set to 223290.
There is nothing wrong with that, but this is larger than 1024,
and could in certain setups cause problems with:
1) software that runs at boot time (e.g., old versions of LILO)
2) booting and partitioning software from other OSs
   (e.g., DOS FDISK, OS/2 FDISK)

Partition type
   p   primary partition (1-4)
   e   extended
Partition number (1-4): First sector (1048576-62521343, default 1048576): Using default value 1048576
Last sector or +size{,K,M,G,T} (1048576-62521343, default 62521343): Command (m for help): Using default value 62521343

The partition table has been altered.
Calling ioctl() to re-read partition table
Waiting for /dev/mmcblk0p1 found with FS
 *   Executing: /usr/libexec/rc/sh/openrc-run.sh /usr/libexec/rc/sh/openrc-run.sh /etc/init.d/modloop start
 *   Mounting modloop  ...
 *   Verifying modloop
 [ ok ]
Waiting for /dev/mmcblk0p2 found with NOFS
Formating data partition on sd card
(1/6) Installing libeconf (0.6.3-r0)
(2/6) Installing libblkid (2.40.2-r4)
(3/6) Installing libcom_err (1.47.1-r1)
(4/6) Installing e2fsprogs-libs (1.47.1-r1)
(5/6) Installing libuuid (2.40.2-r4)
(6/6) Installing e2fsprogs (1.47.1-r1)
Executing busybox-1.37.0-r8.trigger
OK: 14 MiB in 31 packages
mke2fs 1.47.1 (20-May-2024)
Discarding device blocks: done
Creating filesystem with 7684096 4k blocks and 1921360 inodes
Filesystem UUID: fbbc817a-c403-4411-929e-cf1c83b586d0
Superblock backups stored on blocks:
	32768, 98304, 163840, 229376, 294912, 819200, 884736, 1605632, 2654208,
	4096000

Allocating group tables: done
Writing inode tables: done
Creating journal (32768 blocks): done
Writing superblocks and filesystem accounting information: done

Mounting data partition on sd card

Filesystem usage
Filesystem                Size      Used Available Use% Mounted on
devtmpfs                 10.0M         0     10.0M   0% /dev
shm                     453.7M         0    453.7M   0% /dev/shm
tmpfs                   453.7M     15.7M    438.0M   3% /
tmpfs                   181.5M     48.0K    181.4M   0% /run
/dev/mmcblk0p1          511.7M     89.1M    422.6M  17% /media/mmcblk0p1
/dev/loop0               32.9M     32.9M         0 100% /.modloop
/dev/mmcblk0p2           28.7G      2.0M     27.2G   0% /media/mmcblk0p2

Mounts
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,noexec,relatime,size=10240k,nr_inodes=106970,mode=755)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
shm on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on / type tmpfs (rw,relatime,mode=755)
tmpfs on /run type tmpfs (rw,nosuid,nodev,size=185840k,nr_inodes=819200,mode=755)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
tracefs on /sys/kernel/debug/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
/dev/mmcblk0p1 on /media/mmcblk0p1 type vfat (rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,errors=remount-ro)
/dev/loop0 on /.modloop type squashfs (ro,relatime,errors=continue)
/dev/mmcblk0p2 on /media/mmcblk0p2 type ext4 (ro,noatime)

Configuring eth0
 *   Executing: /usr/libexec/rc/sh/openrc-run.sh /usr/libexec/rc/sh/openrc-run.sh /etc/init.d/networking start
 *   Starting networking ...
 *     lo ... [ ok ]
 *     eth0 ...udhcpc: started, v1.37.0
udhcpc: broadcasting discover
udhcpc: broadcasting select for 192.168.0.18, server 192.168.0.1
udhcpc: lease of 192.168.0.18 obtained from 192.168.0.1, lease time 3600
 [ ok ]
Checking network connectivityOK
Setup NTP
 *   service ntpd added to runlevel default
 *   Executing: /usr/libexec/rc/sh/openrc-run.sh /usr/libexec/rc/sh/openrc-run.sh /etc/init.d/ntpd start
 *   Starting busybox ntpd ...
 *   start-stop-daemon: fopen `/run/ntpd.pid': No such file or directory
 *   Detaching to start `/usr/sbin/ntpd' ... [ ok ]
Setup OpenSSH
 *   service sshd added to runlevel default
 *   Caching service dependencies ... [ ok ]
ssh-keygen: generating new host keys: RSA ECDSA ED25519
 *   Starting sshd ...
 *   start-stop-daemon: fopen `/run/sshd.pid': No such file or directory
 *   Detaching to start `/usr/sbin/sshd' ... [ ok ]
Enabling ssh root login
 *   Executing: /usr/libexec/rc/sh/openrc-run.sh /usr/libexec/rc/sh/openrc-run.sh /etc/init.d/sshd stop
 *   Stopping sshd ...
 *   Will stop /usr/sbin/sshd
 *   Will stop PID 2948
 *   Will stop processes of `/usr/sbin/sshd'

 *   Sending signal 15 to PID 2948 ... [ ok ]
 [ ok ]
 [ ok ]
 *   Executing: /usr/libexec/rc/sh/openrc-run.sh /usr/libexec/rc/sh/openrc-run.sh /etc/init.d/sshd start
 *   Starting sshd ...
 *   start-stop-daemon: fopen `/run/sshd.pid': No such file or directory
 *   Detaching to start `/usr/sbin/sshd' ... [ ok ]
Setting keymap to us us
(1/1) Installing kbd-bkeymaps (2.6.4-r2)
OK: 22 MiB in 43 packages
 *   Caching service dependencies ... [ ok ]
 *   Setting keymap ... [ ok ]
(1/1) Purging kbd-bkeymaps (2.6.4-r2)
OK: 21 MiB in 42 packages
Setting up apk cache
Adding repositories and upgrade
Added mirror dl-cdn.alpinelinux.org
Updating repository indexes... done.
Adding myMPDos archiv signing public key
Upgrading all packages
fetch http://dl-cdn.alpinelinux.org/alpine/v3.21/main/aarch64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.21/community/aarch64/APKINDEX.tar.gz
fetch https://raw.githubusercontent.com/jcorporation/myMPDos/master/repository/aarch64/APKINDEX.tar.gz
3.21.0 [/media/mmcblk0p1/apks]
v3.21.0-68-g2357258152c [http://dl-cdn.alpinelinux.org/alpine/v3.21/main]
v3.21.0-75-gd14592b6bd9 [http://dl-cdn.alpinelinux.org/alpine/v3.21/community]
myMPDos [https://raw.githubusercontent.com/jcorporation/myMPDos/master/repository]
OK: 25267 distinct packages available
OK: 21 MiB in 42 packages
Installing packages: mympdos-base mympd mympdos-libmpdclient mympdos-mpc doas alsa-utils rng-tools usbutils raspberrypi busybox-extras net-tools mpg123 mympdos-libgpiod2 mygpiod mympdos-mpd-stable
ERROR: unable to select packages:
  mympdos-base (no such package):
    required by: world[mympdos-base]
Creating user and groups
addgroup: unknown user mympd
Creating files in data partition
install: unknown user mpd
ln: /media/mmcblk0p2/library/USB: No such file or directory
Installing myMPD scripts
'/usr/local/defaults/mympd-scripts/RW-Library.lua' -> '/var/lib/mympd/scripts/RW-Library.lua'
'/usr/local/defaults/mympd-scripts/Reboot.lua' -> '/var/lib/mympd/scripts/Reboot.lua'
'/usr/local/defaults/mympd-scripts/Shutdown.lua' -> '/var/lib/mympd/scripts/Shutdown.lua'
chown: unknown user mympd.mympd
Setting defaults
install: unknown user mpd
cp: can't create '/etc/doas.d/mympd.conf': No such file or directory
Enable automatic reconfiguration of MPD
Enable automount
/usr/local/bin/mympdos-bootstrap.sh: line 414: can't create /etc/mympdos/mympdos.conf: nonexistent directory
/usr/local/bin/mympdos-bootstrap.sh: line 415: can't create /etc/mympdos/mympdos.conf: nonexistent directory
/usr/local/bin/mympdos-bootstrap.sh: line 416: can't create /etc/mympdos/mympdos.conf: nonexistent directory
/usr/local/bin/mympdos-bootstrap.sh: line 417: can't create /etc/mympdos/mympdos.conf: nonexistent directory
/usr/local/bin/mympdos-bootstrap.sh: line 418: can't create /etc/mympdos/mympdos.conf: nonexistent directory
/usr/local/bin/mympdos-bootstrap.sh: line 426: can't create /etc/mympdos/mympdos.conf: nonexistent directory
Configuring MPD
/usr/local/bin/mympdos-bootstrap.sh: line 438: /usr/bin/configmpd.sh: not found
Enabling boot services
 *   rc-update: service `rngd' does not exist
 *   service networking added to runlevel boot
 *   service seedrng added to runlevel boot
 *   rc-update: service `alsa' does not exist
Enabling default services
 *   rc-update: service `mpd' does not exist
 *   rc-update: service `mympd' does not exist
Setting root password
Changing password for root
New password:
Bad password: too weak
Retype password:
passwd: password for root changed by root
Setting usercfg.txt
Removing installation files
Removing wlan packages
OK: 21 MiB in 42 packages
Removing obsolet users and groups
Cleaning up
Create myMPD configuration
/usr/local/bin/mympdos-bootstrap.sh: line 530: mympd: not found
Trusting myMPD CA
cp: can't stat '/var/lib/mympd/ssl/ca.pem': No such file or directory
/usr/local/bin/mympdos-bootstrap.sh: line 534: update-ca-certificates: not found
Setting swclock
Saving configuration
/usr/local/bin/mympdos-bootstrap.sh: line 551: alsactl: not found
Archiving the following files:
etc/
etc/apk/world
etc/apk/arch
etc/apk/keys/alpine-devel@lists.alpinelinux.org-524d27bb.rsa.pub
etc/apk/keys/alpine-devel@lists.alpinelinux.org-58199dcc.rsa.pub
etc/apk/keys/alpine-devel@lists.alpinelinux.org-616a9724.rsa.pub
etc/apk/keys/alpine-devel@lists.alpinelinux.org-616adfeb.rsa.pub
etc/apk/keys/alpine-devel@lists.alpinelinux.org-616ae350.rsa.pub
etc/apk/keys/mail@jcgames.de.rsa.pub
etc/apk/protected_paths.d/lbu.list
etc/apk/repositories
etc/apk/cache
etc/runlevels/
etc/runlevels/default/
etc/runlevels/default/local
etc/runlevels/default/ntpd
etc/runlevels/default/sshd
etc/runlevels/sysinit/devfs
etc/runlevels/sysinit/dmesg
etc/runlevels/sysinit/mdev
etc/runlevels/sysinit/hwdrivers
etc/runlevels/sysinit/modloop
etc/runlevels/boot/modules
etc/runlevels/boot/sysctl
etc/runlevels/boot/hostname
etc/runlevels/boot/bootmisc
etc/runlevels/boot/syslog
etc/runlevels/boot/swclock
etc/runlevels/boot/loadkmap
etc/runlevels/boot/networking
etc/runlevels/boot/seedrng
etc/runlevels/shutdown/mount-ro
etc/runlevels/shutdown/killprocs
etc/runlevels/shutdown/savecache
etc/conf.d/
etc/conf.d/local
etc/conf.d/loadkmap
etc/local.d/
etc/fstab
etc/group
etc/hostname
etc/inittab
etc/passwd
etc/shadow
etc/network/interfaces
etc/lbu/lbu.conf
etc/mdev.conf
etc/resolv.conf
etc/zoneinfo/
etc/zoneinfo/UTC
etc/localtime
etc/ssh/sshd_config
etc/ssh/ssh_host_rsa_key.pub
etc/ssh/ssh_host_rsa_key
etc/ssh/ssh_host_ecdsa_key.pub
etc/ssh/ssh_host_ecdsa_key
etc/ssh/ssh_host_ed25519_key.pub
etc/ssh/ssh_host_ed25519_key
etc/keymap/
etc/keymap/us.bmap.gz
etc/mympdos/
etc/mympdos/custom/
etc/mympdos/custom/mpd.replace
etc/passwd-
etc/shadow-
etc/group-
var/lib/misc/openrc-shutdowntime
var/lib/mympd/
var/lib/mympd/scripts/
var/lib/mympd/scripts/RW-Library.lua
var/lib/mympd/scripts/Reboot.lua
var/lib/mympd/scripts/Shutdown.lua
Created /media/mmcblk0p2/myMPDos.apkovl.tar.gz
Successfully saved apk overlay files
```

**Comment by jcorporation** [comment]
Thanks for reporting this issue.

The mympdos-base package is not in the repository. I added it again, it was a build failure with Alpine Linux 3.21.

I also released a 1.6.1 release fixing the regressions introduces by the upgrade to Alpine Linux 3.21.

**Comment by keithmillar** [comment]
Thanks For the fast update.
The software now appears to be installed, however no sound cards are installed so I can only play back in a browser and not locally on the pi.
